src/panels/lovelace/components/hui-generic-entity-row.js

```diff
@@ -70,6 +73,11 @@ class HuiGenericEntityRow extends PolymerElement {
                 ></ha-relative-time>
               </template>
             </template>
+            <template is="dom-if" if="[[!config.secondary_info]]">
```

Comment:
This is a bad check as it means by default we're not showing the secondary line, as the config will be undefined.

Reply:
Isn't that what we want? We could use a method to return `config && config.secondary_info`, but wouldn't that would mean the `<slot>` is always attached first, then once `config` gets set, if `secondary_info` is defined, the slot gets removed and one of the others gets attached? The current way, we create the fragments, but don't attach anything to the dom until `config` is defined

Reply:
hmm. I actually am thinking now that this shouldn't be configured via `config`, as that's user input?

Reply:
Ok, now you lost me :laughing:

Reply:
haha. Isn't `config` set by users in lovelace cards? But the slot content is filled by the media player row, so shouldn't be configurable by users?

Reply:
Ohh I see what you're saying. The slot is for the row to show some optional default content, but all the different row types allow users to override `secondary_info` and show either `last_changed` or `entity_id` so I think it still must be configurable for the override

Reply:
Ah I see. In that case I would still want it not be extracted from the config by the generic row, but instead the component that renders generic-row can have a `show-secondary=[[xx]]`, such that we don't depend on a config implementation, as for example for media player we might always want to show it.

Reply:
Although it might not be necessary since we can use CSS [`::slotted`](https://developer.mozilla.org/en-US/docs/Web/CSS/::slotted) to achieve similar things?

Reply:
Actually… we should never pass a slot into a component and then pass a variable to not render the slot. Instead, we should have never filled the slot to begin with.

Reply:
Sorry I'm not sure I follow. Should it just be up to each row type to choose whether or not to fill the secondary slot and then inform the generic row? The secondary info needs to be within the `.info` div in the generic row's shadow root for the flex layout to work. It seems like the generic row still needs to check `config.secondary_info` to show last-changed or entity_id, or should we move all that logic to each individual row class?

Reply:
And based on your comments, isn't the fact that the entire info template and primary slot is conditional on `_stateObj` an issue as well? That's kinda the same situation since we always fill that slot in the rows, but the generic row determines whether or not render it?

Reply:
That makes sense to me in general, just not in the context of generic row. The way it's structured now, generic row has to know about config. It's the generic row that determines the state display, the icon, the secondary-info, and whether or not to even display anything passed in from each row type in the main `<slot>` based on whether or not it can extract a `stateObj`. Was this the wrong way to build it from the start?

For example: The text and timer rows always insert a div in the main `<slot>`, but the generic row might not render it based on its own `_computeStateObj` method.

Reply:
I think that if a stateObj is passed in, it is fine for generic-row to render "unavailable" if that value is null.

The problem with having the generic-row read the config for what to render in secondary, is when a domain wants to have a different default for secondary option based on some other setting? Now we're stuck :) 

Choosing abstractions is hard. We can go very specific and save a lot of code or go very flexible and barely abstract anything at all. As features grow, it's good to reconsider our abstractions and to ensure we're not working against ourselves.

With this newly added if-statement. We would now have to test it with every possible permutation to make sure we didn't break anything?

Reply:
Okay I think I'm with you. I think part of my confusion was because the if-statement I added was the "else" to an existing condition, but I guess it's only important that the rows have the option to render something different in secondary-info independent of the generic row. 

So the whole if-else based on `config.secondary_info` needs to use a value that comes from the specific rows instead? Then they can set a prop on generic row if they aren't going to use the slot so generic row can decide to render something else